### PR TITLE
Add Rec Seq2seq components and configure file, change all config files due to interface change

### PR DESCRIPTION
**Pull request by zhtmike**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/rec/crnn/crnn_resnet34.yaml [hunk 1]
```diff
@@ -185,7 +186,7 @@ predict:
       - ToCHWImage:
     #  the order of the dataloader list, matching the network input and the input labels for the loss function, and optional data for debug/visaulize
     output_columns: [ 'img_path', 'image', 'raw_img_shape' ]
-    num_columns_to_net: 1 # num inputs for network forward func
+    columns_indices_for_net: [0] # input indices for network forward func in output_columns
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_resnet34.yaml:189]: img_path 作为网络输入看起来有些问题

File: configs/rec/crnn_seq2seq/crnn_seq2seq_resnet34.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          padding: False # aspect ratio will be preserved if true.
+      - NormalizeImage: # different from paddle (paddle wrongly normalize BGR image with RGB mean/std from ImageNet for det, and simple rescale to [-1, 1] in rec.
+          bgr_to_rgb: True
+          is_hwc: True
+          mean: [127.0, 127.0, 127.0]
+          std: [127.0, 127.0, 127.0]
+      - ToCHWImage:
+    #  the order of the dataloader list, matching the network input and the input labels for the loss function, and optional data for debug/visaulize
+    output_columns: ["image", "text_seq"] #, 'length'] #'img_path']
+    columns_indices_for_net: [0, 1] # input indices for network forward func in output_columns
```
Comment (SamitHuang) [inline configs/rec/crnn_seq2seq/crnn_seq2seq_resnet34.yaml:96]: 这里不用指定column_indices_for_label?

File: mindocr/data/transforms/rec_transforms.py [hunk 3]
```diff
@@ -280,3 +377,13 @@ def __call__(self, data):
     assert trans.num_classes==96
     assert out['length'] ==len(text), 'Not equal: {}, {}'.format(out['length'], text) # use_space_char=False, but the dict contains space, % is also in dict
     print(seq)
```
Comment (SamitHuang) [inline mindocr/data/transforms/rec_transforms.py:379]: 这些中间的print是否必要? 如仅调试用，应删去。

File: mindocr/models/base_model.py [hunk 4]
```diff
@@ -33,13 +33,16 @@ def __init__(self, config: dict):
 
         self.model_name = f'{backbone_name}_{neck_name}_{head_name}'
 
-    def construct(self, x):
+    def construct(self, x, y=None):
```
Comment (SamitHuang) [inline mindocr/models/base_model.py:36]: 需要补充说明入参，x和y。并更新mindocr/models/README.md相应的接口描述。

File: README.md
Comment (SamitHuang) [inline README.md:1]: 建议优化命名：
columns_indices_for_net -> net_input_column_index
columns_indices_for_label -> label_column_index
columns_indices_for_meta_data -> meta_data_column_index
另外，meta_data_column_index 应避免手动指定，output_columns中除去net_input_column_index 和 label_column_index部分剩下都是meta data.

File: configs/rec/crnn/crnn_resnet34.yaml (hunk 1 above)
Comment (zhtmike) [reply configs/rec/crnn/crnn_resnet34.yaml]: 经确认pred pipeline目前是不会用到num_columns_to_net的，所以这个参数配置已删除。

File: configs/rec/crnn_seq2seq/crnn_seq2seq_resnet34.yaml (hunk 2 above)
Comment (zhtmike) [reply configs/rec/crnn_seq2seq/crnn_seq2seq_resnet34.yaml]: column_indices_for_label如果不提供会 implicitly 用indices = [1]做为label。现已添加。

File: mindocr/data/transforms/rec_transforms.py (hunk 3 above)
Comment (zhtmike) [reply mindocr/data/transforms/rec_transforms.py]: 已删去。

File: README.md
Comment (zhtmike) [reply README.md:1]: 已更新命名。meta_data_column_index已按这逻辑更新。

File: tools/train.py [hunk 5]
```diff
@@ -147,9 +150,9 @@ def main(cfg):
         batch_size=cfg.train.loader.batch_size,
         ckpt_save_dir=cfg.train.ckpt_save_dir,
         main_indicator=cfg.metric.main_indicator,
-        ema=ema,
```
Comment (hadipash) [inline tools/train.py:150]: lost `ema` argument

File: mindocr/models/base_model.py (hunk 4 above)
Comment (hadipash) [reply mindocr/models/base_model.py:40]: I wonder how we will deal with the problem if the backbone or neck will require an extra input.

**Review by hadipash — CHANGES_REQUESTED** [review]

File: mindocr/losses/builder.py [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import inspect
+from mindspore.nn import CrossEntropyLoss
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply mindocr/losses/builder.py]: cleaned.

File: mindocr/utils/callbacks.py [hunk 7]
```diff
@@ -83,9 +88,15 @@ def eval(self):
...
+                inputs = [data[x] for x in self.input_indices]
+            else:
+                inputs = [data[0]]
 
-            inputs = data[:self.num_inputs] # [imgs]
-            gt = data[self.num_inputs:] if self.num_labels is None else data[self.num_inputs: self.num_inputs+self.num_labels]
+            if self.label_indices is not None:
+                gt = [data[x] for x in self.label_indices]
+            else:
+                gt = [data[1]]
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply mindocr/utils/callbacks.py:99]: The rest of the axes will be marked as meta_data.

File: tools/train.py (hunk 5 above)
Comment (zhtmike) [reply tools/train.py:150]: fixed.

File: configs/det/dbnet/db_r50_icdar15.yaml [hunk 8]
```diff
@@ -143,8 +144,8 @@ eval:
       - ToCHWImage:
     #  the order of the dataloader list, matching the network input and the labels for evaluation
     output_columns: [ 'image', 'polys', 'ignore_tags' ]
-    num_columns_to_net: 1 # num inputs for network forward func
-    num_columns_of_labels: 2 # num labels
+    net_input_column_index: [0] # input indices for network forward func in output_columns
+    label_column_index: [1, 2] # input indices marked as label
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply configs/det/dbnet/db_r50_icdar15.yaml:148]: yes, they have default value. net_input_column_index is default to be [0], label_column_index is default to [1]. We can create a PR later to clean the all unecessary settings.

File: mindocr/utils/callbacks.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (hadipash) [reply mindocr/utils/callbacks.py:99]: Can it be `gt = [data[start_idx:end_idx]]` where `end_idx` is calculated based on `self.meta_data_indices` the same way as `start_idx` does?

File: mindocr/models/base_model.py (hunk 4 above)
Comment (zhtmike) [reply mindocr/models/base_model.py:40]: Current we don't support this feature.

File: mindocr/utils/callbacks.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (hadipash) [reply mindocr/utils/callbacks.py:99]: Can it be `gt = [data[start_idx:end_idx]]` where `end_idx` is calculated based on `self.meta_data_indices` the same way as `start_idx` does?

**Review by hadipash — APPROVED** [review]

File: mindocr/utils/callbacks.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (hadipash) [reply mindocr/utils/callbacks.py:99]: We can assume that the indices are continuous for the default case, so we can avoid putting default values into config files.

**Review by SamitHuang — APPROVED** [review]
Good job! Some things to be supplemented next 
1.  readme file for crnn_seq2seq.
2.  avoid setting meta_data_indices as input for simplicity, this can be inferred by set(all_indices) - set(input_indices) - set(label_indices).

**Pull request merged**